### ISSUE TITLE
origin_upper_left and gl_FragCoord in ESSL

### ISSUE THREAD

**Issue by mlomb** [LukasBanana/XShaderCompiler#67]
I had some problems compiling some simple shaders into ESSL.

Input shader:
```
struct Input {
    float4 position : SV_POSITION;
};

float4 main(Input input) : SV_TARGET{
    return float4(input.position.x / 1000, 0, 0, 1);
}
```

Output shader:
```
#version 300 es

layout(origin_upper_left) in vec4 gl_FragCoord;

out highp vec4 SV_Target0;

void main()
{
    SV_Target0 = vec4(gl_FragCoord.x / 1000.0, 0, 0, 1);
}
```
The errors I get:

- 'origin_upper_left' : invalid layout qualifier
- '' : No precision specified for (float)
- 'gl_' : reserved built-in name

compiling with:
OpenGL version: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Shading version: OpenGL ES GLSL 1.00 (WebGL)

I think the cause of the errors are:

-  layout(origin_upper_left) is not supported in ESSL
- `in vec4 gl_FragCoord` doesn't have a precision specified. i.e.  `in vec4 highp gl_FragCoord`
- ESSL doesn't allow to use 'gl_' for a variable name. 

3.9 of the specification:
> Identifiers starting with “gl_” are reserved for use by OpenGL ES, and may not be declared in a shader as
> either a variable or a function. It is an error to redeclare a variable, including those starting “gl_”.

And something like this is what I was expecting:
```
#version 300 es

in highp vec2 position;

out highp vec4 SV_Target0;

void main()
{
    SV_Target0 = vec4(position.x / 1000.0, 0, 0, 1);
}
```

GLSL ES 3.00 Specification: https://www.khronos.org/registry/OpenGL/specs/es/3.0/GLSL_ES_Specification_3.00.pdf

**Comment by LukasBanana** [comment]
This is a redeclaration which is allowed in GLSL, but as it seems not for ESSL.

By the way, your expected shader uses a wrong input field: `position` was declared with the `SV_POSITION` semantic which must be translated to `gl_FragCoord` when a fragment shader is translated into GLSL.
This is because in a vertex shader `SV_POSITION` maps to `gl_Position` while in a fragment shader it maps to `gl_FragCoord`.
However, the origin of screen coordinates is different between OpenGL and Direct3D.
(0,0) is **lower-left** in OpenGL, while (0,0) is **upper-left** in Direct3D.
As a result, `gl_FragCoord.y` is not the same as `position.y`.
Although the Y-axis is not used in your shader, this is a common problem.

If such a redeclaration is not supported in ESSL, I can only remove it from the ESSL output so far.

**Comment by LukasBanana** [comment]
Fixed with [72d4f9a](https://github.com/LukasBanana/XShaderCompiler/commit/72d4f9ac3a6a08c41e83f92c8d0d851b45eef38d).
Thanks for the report.

I checked the ESSL 3.2 spec and indeed it looks like there is no way to change the origin of fragment coordinates :-/
